### DISCUSSION TITLE
Hantek6022BL - SW for the logic channels?

### DISCUSSION THREAD

**Discussion by tudopatroi** [OpenHantek/OpenHantek6022#179] [Q&A] [answered]
Hi guys,
I am new here, so please bear with me if the answer is already here but I could not find it.
Got today an Hantek6022BL, tried for a bit the original SW just to confirm what everyone was saying that it is horrible and does not work (I am an optimist, hoped they fixed it by now).
Then I got it working with the OpenHantek6022 (20210408 build 867) (FW0209). All would be great but I only see the two main channels.
Do I need a different SW to see the other 16 digital inputs? Are there any plans to add them to this one?
Thank you,

**Comment by Ho-Ro** [comment, marked as answer]
> Do I need a different SW to see the other 16 digital inputs? 
> Are there any plans to add them to this one?

Short answer: OpenHantek is a DSO (digital storage oscilloscope) program - it is not an LA (logic analyser) tool. 
If you want to use the LA part, then [sigrok](https://sigrok.org) is the way to go. You didn't mention your setup - but anyway, sigrok works (besides Linux) also for Windows and IIRC also for macOS. The [Hantek6022BL](https://sigrok.org/wiki/Hantek_6022BL) can either be used as oscilloscope or as logic analyzer, but not both at the same time. I.e., it is not a mixed-signal-oscilloscope (MSO). You choose one of the two functions with the switch on the back, so there is no point in supporting the LA input from OpenHantek.

If you want more info about the 6022 devices take two or three hours and read the [monster thread](https://www.eevblog.com/forum/testgear/hantek-6022be-20mhz-usb-dso/) from down-under.

**Reply by tudopatroi** [reply]
Thank you very much, I am on W10 normally, I will try sigrok.

**Reply by deconstructo** [reply]
sigrok works on macOS

**Reply by hramrach** [reply]
I think this would be nice to document more clearly.

**Reply by Ho-Ro** [reply]
@hramrach good hint, done in a83f35f

**Reply by hramrach** [reply]
Looks good, thanks.
